Libraries/Opc.Ua.Client/Subscription.cs

```diff
@@ -670,15 +664,7 @@ public uint SequenceNumber
         {
             get
             {
-                lock (m_cache)
-                {
-                    if (m_messageCache.Count > 0)
-                    {
-                        return m_messageCache.Last.Value.SequenceNumber;
-                    }
-                }
-
-                return 0;
+                return m_messageCache.LastOrDefault()?.SequenceNumber ?? 0;
```

Comment:
the peek is at the wrong end here. It seems the concurrent queue is not so suited for the `Last` operations and I rather revert to the LinkedList to avoid regressions.